src/lib/checkbox/checkbox.ts

```diff
@@ -193,16 +188,22 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
     super(elementRef);
 
     this.tabIndex = parseInt(tabIndex) || 0;
-  }
 
-  ngAfterViewInit() {
-    this._focusMonitor
-      .monitor(this._inputElement.nativeElement)
-      .subscribe(focusOrigin => this._onInputFocusChange(focusOrigin));
+    this._focusMonitor.monitor(elementRef.nativeElement, true).subscribe(focusOrigin => {
+      // TODO(paul): support `program`. See https://github.com/angular/material2/issues/9889
```

Comment:
You should be able to remove this TODO since we no longer control the focus ripple based on the `FocusMonitor`.